### PR TITLE
Remove tutorials from readthedocs

### LINKED ISSUE #2347: Remove tutorials from readthedocs
they're getting out of date, and I don't think anyone is using them

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
# Description

This removes the tutorials for adding models, solvers, and spatial methods from PyBaMM's documentation.

Fixes #2347 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ flake8`
- [x] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by agriyakhetarpal** [comment]
I could not find any other references to the tutorials besides the `index.rst` file, but going through the documentation brought to my attention some missing links, a few typos (mostly the same ones across methods, therefore most likely a single typo that got copied over) and missing references to methods within classes. Would it be preferred to open separate PRs for each type?

**Comment by tinosulzer** [comment]
Thanks, feel free to add to this PR

**Comment by agriyakhetarpal** [comment]
This is probably unneeded, but there are some inconsistencies in spelling schema: for example; certain words that appear extensively in the documentation could be spellchecked and case-corrected:

python could be renamed to Python, casadi to CasADi, neumann to Neumann, pybamm to PyBaMM, laplacian to Laplacian, bibtex to BibTeX, and so on.

File: pybamm/expression_tree/unary_operators.py [hunk 1]
```diff
@@ -634,7 +634,7 @@ def __init__(self, child, integration_variable):
         if isinstance(integration_variable, list):
             if len(integration_variable) > 1:
                 raise NotImplementedError(
-                    "Indefinite integral only implemeted w.r.t. one variable"
+                    "Indefinite integral only implemented w.r.t. one variable"
```
Comment (Saransh-cpp) [inline pybamm/expression_tree/unary_operators.py:637]: For the [failing tests](https://github.com/pybamm-team/PyBaMM/actions/runs/3669113413/jobs/6202712976) -

The errors and warnings are tested using `unittest` in the `tests` directory. Changing the error message should be accompanied by making the same change in the tests. For this case, here -

https://github.com/pybamm-team/PyBaMM/blob/c38f385675c0a6481de776acd5de7182eca77ab8/tests/unit/test_expression_tree/test_unary_operators.py#L357-L360

Comment (agriyakhetarpal) [reply pybamm/expression_tree/unary_operators.py:637]: I had missed that. Thanks

**Comment by agriyakhetarpal** [comment]
Some other typos: '**a implemented**' – to be corrected to '**as implemented**' in [Potential Pair models](https://pybamm.readthedocs.io/en/latest/source/models/submodels/current_collector/potential_pair.html). The same typo exists across multiple methods across [Submodels](https://pybamm.readthedocs.io/en/latest/source/models/submodels/index.html).

I was not able to find where to correct these, however

**Comment by tinosulzer** [comment]
It's in `base_submodel`. If you use VS Code, you can do whole-directory search with control+shift+F

**Comment by tinosulzer** [comment]
> What should be done here?

Could you just make them full links to the github?

> python could be renamed to Python, casadi to CasADi, neumann to Neumann, pybamm to PyBaMM (except for folder names), laplacian to Laplacian, jacobian to Jacobian, bibtex to BibTeX, and so on.

This would be pretty neat if you can figure out how to do it automatically

**Comment by tinosulzer** [comment]
Yes that is the best way I can think of, especially since if reuses an existing workflow

**Comment by Saransh-cpp** [comment]
You can do this using VSCode, but you will need to ensure that it does not edit code occurrences (opening the `docs` folder as the root directory inside VSCode should eliminate almost all of the code) -

image

**Comment by agriyakhetarpal** [comment]
I did think of it, but it could mess up autoclasses wherein these words make up the name of methods?

**Comment by Saransh-cpp** [comment]
Yes, would mess up a lot of things 😬

**Comment by agriyakhetarpal** [comment]
I can fix them manually for now if that would be fine

**Review by Saransh-cpp — APPROVED** [review]
Thanks, @agriyakhetarpal! Is this ready for merging?

**Comment by agriyakhetarpal** [comment]
Yes, please go ahead!

**Comment by tinosulzer** [comment]
Thanks!

**Comment by tinosulzer** [comment]
@all-contributors add @agriyakhetarpal for docs

**Pull request merged**